cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -104,25 +90,27 @@ func main() {
 					continue
 				}
 
-				// Checks if scheduler tried to schedule the pods after thew newest node was added.
-				newestNode := GetNewestNode(nodes)
-				if newestNode == nil {
-					glog.Errorf("No newest node")
-					continue
-				}
-				oldestSchedulingTrial := GetOldestFailedSchedulingTrail(pods)
-				if oldestSchedulingTrial == nil {
-					glog.Errorf("No oldest unschedueled trial: %v", err)
+				pods, err := unschedulablePodLister.List()
+				if err != nil {
+					glog.Errorf("Failed to list unscheduled pods: %v", err)
 					continue
 				}
 
-				// TODO: Find better way to check if all pods were checked after the newest node
-				// was added.
-				if newestNode.CreationTimestamp.After(oldestSchedulingTrial.Add(-1 * time.Minute)) {
-					// Lets give scheduler another chance.
-					glog.V(1).Infof("One of the pods have not been tried after adding %s", newestNode.Name)
+				// We need to reset all pods that have been marked as unschedulable before newest
```

Comment:
nit: AFAIU also pods marked as unschedulable in the same time the node became available are reset here, so to be super precise the comment should be updated.